soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -222,9 +237,9 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		z_xt_ints_off(0xffffffff);
 		core_desc[cpu].bctl = DSPCS.bootctl[cpu].bctl;
 		DSPCS.bootctl[cpu].bctl &= ~DSPBR_BCTL_WAITIPCG;
-		soc_cpus_active[cpu] = false;
-		z_xtensa_cache_flush_inv_all();
 		if (cpu == 0) {
+			soc_cpus_active[cpu] = false;
```

Comment:
Won't this now fail to update soc_cpus_active[] for secondare cores?

Reply:
No. For secondary core its done inside of _power_gate_entry_ function.